src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/resource/CourtMaintenanceResource.kt

```diff
@@ -68,8 +73,11 @@ class CourtMaintenanceResource(private val courtService: CourtService) {
     @Schema(description = "Court ID", example = "ACCRYC", required = true)
     @PathVariable @Size(max = 12, min = 2, message = "Court ID must be between 2 and 12") courtId: String,
     @RequestBody @Valid courtUpdateRecord: UpdateCourtDto
-  ): CourtDto =
-    courtService.updateCourt(courtId, courtUpdateRecord)
+  ): CourtDto {
+    val updateCourt = courtService.updateCourt(courtId, courtUpdateRecord)
+    snsService.sendEvent(COURT_REGISTER_UPSERT, courtId)
```

Comment:
would separate events be better? One for update one for insert ????

Reply:
I wonder that too, I suppose it makes it easier to sync